eventmesh-registry-plugin/eventmesh-registry-consul/src/main/java/org/apache/eventmesh/registry/consul/service/ConsulRegistryService.java

```diff
@@ -0,0 +1,174 @@
+                    }
+                    this.consulHost = addr[0];
+                    this.consulPort = addr[1];
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public void start() throws RegistryException {
+        consulClient = new ConsulClient(new ConsulRawClient(consulHost, Integer.parseInt(consulPort), getHttpClient()));
+    }
+
+    @Override
+    public void shutdown() throws RegistryException {
+        INIT_STATUS.compareAndSet(true, false);
+        START_STATUS.compareAndSet(true, false);
+        consulClient = null;
+    }
+
+    @Override
+    public boolean register(EventMeshRegisterInfo eventMeshRegisterInfo) throws RegistryException {
+        try {
+            String[] ipPort = eventMeshRegisterInfo.getEndPoint().split(IP_PORT_SEPARATOR);
+            NewService service = new NewService();
+            service.setPort(Integer.parseInt(ipPort[1]));
+            service.setAddress(ipPort[0]);
+            service.setName(eventMeshRegisterInfo.getEventMeshClusterName());
+            service.setId(eventMeshRegisterInfo.getEventMeshClusterName() + "-" + eventMeshRegisterInfo.getEventMeshName());
+            consulClient.agentServiceRegister(service);
+        } catch (Exception e) {
+            throw new RegistryException(e.getMessage());
+        }
+        logger.info("EventMesh successfully registered to consul");
+        return true;
+    }
+
+    @Override
+    public boolean unRegister(EventMeshUnRegisterInfo eventMeshUnRegisterInfo) throws RegistryException {
+        try {
+            consulClient.agentServiceDeregister(eventMeshUnRegisterInfo.getEventMeshClusterName() + "-" + eventMeshUnRegisterInfo.getEventMeshName());
+        } catch (Exception e) {
+            throw new RegistryException(e.getMessage());
+        }
+        logger.info("EventMesh successfully unregistered to consul");
+        return true;
+    }
+
+    @Override
+    public List<EventMeshDataInfo> findEventMeshInfoByCluster(String clusterName) throws RegistryException {
+        Map<String, Service> agentServices = consulClient.getAgentServices().getValue();
+        HealthServicesRequest request = HealthServicesRequest.newBuilder().setPassing(true).build();
+        consulClient.getHealthServices(clusterName, request);
+        List<EventMeshDataInfo> eventMeshDataInfos = new ArrayList<>();
+        agentServices.forEach((k, v) -> {
+            String[] split = v.getId().split("-");
+            eventMeshDataInfos.add(new EventMeshDataInfo(split[0], split[1], v.getAddress() + ":" + v.getPort(), 0));
+        });
+        return eventMeshDataInfos;
+    }
+
+    @Override
+    public Map<String, Map<String, Integer>> findEventMeshClientDistributionData(String clusterName, String group, String purpose)
+        throws RegistryException {
+        return Collections.emptyMap();
+    }
+
+
+    private HttpClient getHttpClient() {
```

Comment:
consul already provides DEFAULT_HTTP_TRANSPORT, I don't think this code is necessary

Reply:
yep,I will delete it